### ISSUE TITLE
False positive IDE0004 and false negative CS0121 reported by the analyzer when calling certain kind of method overloads

### ISSUE THREAD

**Issue by RedSaltyFish** [dotnet/roslyn#72933]
**Version Used**: 4.10.0-2.24124.2 (3059a073)

**Steps to Reproduce**:

1. Define two overloads as shown below.
```
void Foo(char[] input)
{
    Console.WriteLine(input);
}

void Foo(string input)
{
    Console.WriteLine(input);
}
```

2. Call the defined method.
```
Foo((char[])['h', 'e', 'l', 'l', 'o']);
```

**Diagnostic Id**: False positive `IDE0004`, false negative `CS0121` if `(char[])` conversion is removed.

**Expected Behavior**: Analyzer doesn't produce `IDE0004` when the code is written as shown above and detects `CS0121` when `(char[])` conversion is removed.

**Actual Behavior**: False positive `IDE0004` reported by analyzer suggesting `(char[])` conversion is redundant. Remove `(char[])` conversion and no `CS0121` is reported by analyzer, but compiling it results in a `CS0121` failure.

**Comment by CyrusNajmabadi** [comment]
I don't repro this:

![image](https://github.com/dotnet/roslyn/assets/4564579/1e09ee05-9b6c-4fdf-b983-8c1e1e6a9a37)

@cston @RikkiGibson did something change here?

**Comment by RikkiGibson** [comment]
Yes I think we now require presence of an applicable Add method for non-empty collection-exprs in order for the conversion to exist, which causes the string overload to be non-applicable.

**Comment by CyrusNajmabadi** [comment]
@RedSaltyFish this should go away once you upgrade to a more recent version of C# (which may or may not have been released yet :)).
